dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -344,23 +347,58 @@ def read_metadata(
         # Create `parts`
         # This is a list of row-group-descriptor dicts, or file-paths
         # if we have a list of files and gather_statistics=False
+        base_path = (base_path or "") + fs.sep
         if not parts:
+            # Populate `partsin` with dataset row-groups
             partsin = pf.row_groups
-            if fast_metadata:
-                pf = (paths, gather_statistics)
+            partitions = pf.info.get("partitions", None)
+            if partitions and not fast_metadata:
+                # We have partitions, and do not have
+                # a "_metadata" file for the worker to read.
+                # Therefore, we need to pass the pf object in
+                # the task graph
+                pf_deps = pf
```

Comment:
There was some code to strip down the pf instance to avoid thrift serialisation costs. Is that still happening?